src-electron/upgrade/upgrade.js

```diff
@@ -40,14 +48,48 @@ async function getForcedExternalStorage(db, attributeId) {
   let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
   let zcl = await queryPackage.getPackageByPackageId(db, pkgs)
   zcl = zcl.path
-  let obj = await fsp.readFile(zcl)
-  let data = JSON.parse(obj)
+  let obj = await fsp.readFile(zcl, 'utf-8')
+  let data
+  if (isJsonString(obj)) {
+    data = JSON.parse(obj)
+  }
   let byName = data?.attributeAccessInterfaceAttributes
   let lists = data?.listsUseAttributeAccessInterface
   let forcedExternal = { byName, lists }
   return forcedExternal
 }
 
+/**
+ * Returns a list of attributes with proper global attributes
```

Comment:
This is ... not very helpful.  Please document what this function _actually_ does.

Reply:
will do, thanks

Reply:
added better comments and will keep working on them